core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -237,6 +238,12 @@ public static String initcap(String s) {
     return newS.toString();
   }
 
+  /** SQL ASCII(string) function. */
+  public static int ascii(String s) {
+    return s.length() == 0
+        ? 0 : s.substring(0, 1).getBytes(StandardCharsets.UTF_8)[0];
```

Comment:
why not use [charAt()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#charAt-int-) ?

Reply:
Substring returns a string which has getBytes function and it can specify charset.

Reply:
I did a little test:
```java
  @Test
  public void basic() {
    final char[] chars = {0x00, 0x1B,
        'A', 'a', '0', '{', '}',
        '\n', '\r', '\t', ' ',
        0x80, 0x99,
        '\u0391', '\u03A9', '\u0391'};

    for (char ch: chars) {
      final String value = String.valueOf(ch);
      assertEquals(String.format("for %s (ascii:%d hex:0x%02X)", value, (int) ch, (int) ch),
          ascii1(value), ascii2(value));
    }
  }

  private static int ascii1(String str) {
    return str.isEmpty() ? 0 : str.getBytes(StandardCharsets.UTF_8)[0];
  }

  private static int ascii2(String str) {
    return str.isEmpty() ? 0 : str.charAt(0);
  }
```
So it is failing starting with extended ascii characters (> 127)  :  
```
java.lang.AssertionError: for  (ascii:128 hex:0x80) 
Expected :-62
Actual   :128
```
It seems for standard ascii table (hex: 0x00 - 0x7E) `charAt()` produces identical results as UTF8 encoding. It is expected because UNICODE is identical to ASCII for first 128 characters (and UTF8 uses single byte for basic ASCII charset).

The question is how to define Calcite  `ASCII` function outside basic ASCII chars (0-127) ? Or explicitly say behaviour is undefined for non-basic ascii ? 

[Transact-SQL](https://docs.microsoft.com/en-us/sql/t-sql/functions/ascii-transact-sql?view=sql-server-2017) says ASCII works only for [printable chars](https://en.wikipedia.org/wiki/ASCII#Printable_characters) (0x20- 0x7E)

Reply:
It looks like the behavior is vary from database system to database system.  Such as:

[PostgreSQL](http://www.postgresqltutorial.com/postgresql-ascii/) says the ASCII() function returns the Unicode code point of the character in the case of UTF-8.

Reply:
Yes. We usually follow Postgres standards (eg. `extract` function). 
In this case is `charAt()` is preferable ?

Reply:
Updated the PR in which I use `codePointAt` to get the Unicode code point of the character. 

Is it proper?

Reply:
Yes it is more generic than `charAt()`.